core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -481,6 +492,14 @@ class CivilizationInfo {
             for(tradeRequest in diplomacyManager.otherCiv().tradeRequests.filter { it.requestingCiv==civName })
                 diplomacyManager.otherCiv().tradeRequests.remove(tradeRequest) // it  would be really weird to get a trade request from a dead civ
         }
+
+        if (isMapRevealEnabled()) {
+            exploredTiles = gameInfo.tileMap.values.map { it.position }.toHashSet()
+        }
+
+        if (isCurrentPlayer()) {
+            UncivGame.Current.setScreen(VictoryScreen())
+        }
```

Comment:
I think we should try to decouple as much as possible the UI from the backend. Do we really need to set the victory screen in the civ info destroy method?

Reply:
Agree, UI should not be referenced anywhere from the Models

Reply:
Also the 'map reveal enable' is a UI thing and not a Civ thing, so this should be checked in WorldMapHolder probably, and not in CivilizationInfo